src/edu/stanford/nlp/ie/NERFeatureFactory.java

```diff
@@ -397,68 +396,180 @@ public void init(SeqClassifierFlags flags) {
    */
   @Override
   public Collection<String> getCliqueFeatures(PaddedList<IN> cInfo, int loc, Clique clique) {
-    Collection<String> features = Generics.newHashSet();
+    Set<String> features = Generics.newHashSet(100);
+    FeatureCollector c = new FeatureCollector(features);
```

Comment:
I wish there were some way to specify that FeatureCollector needs to have the suffix set at least once.  Perhaps the initial suffix could be part of the constructor?  Not sure if that would actually be cleaner.

Reply:
The suffix is valid for a context; I'd even rather un-set it afterwards to avoid programming errors.
It would likely be nicer to move the setSuffix call from
```
featuresCpC(cInfo, loc, c.setSuffix("CpC"));
```
into the beginning of method `featuresCpC`, and also add a `c.setSuffix(null);` to the end to make it an invalid value. But there is at least one method which is called twice with a different suffix and slightly different parameters.